xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h

```diff
@@ -645,7 +1015,129 @@ inline void SetSettingFloat(const std::string& settingName, float settingValue)
   CAddonBase::m_interface->toKodi->set_setting_float(CAddonBase::m_interface->toKodi->kodiBase,
                                                      settingName.c_str(), settingValue);
 }
-//----------------------------------------------------------------------------
+//------------------------------------------------------------------------------
+
+//==============================================================================
+/// @brief Check and get a enum setting value.
+///
+/// The setting name relate to names used in his <b>settings.xml</b> file.
+///
+/// @param[in] settingName The name of asked setting
+/// @param[out] settingValue The given setting value
+/// @return true if setting was successfully found and "settingValue" is set
+///
+/// @remark The enums are used as integer inside settings.xml.
+/// @note If returns false, the "settingValue" is not changed.
+///
+///
+/// ----------------------------------------------------------------------------
+///
+/// **Example:**
+/// ~~~~~~~~~~~~~{.cpp}
+/// #include <kodi/General.h>
+///
+/// enum myEnumValue
+/// {
+///   valueA,
+///   valueB,
+///   valueC
+/// };
+///
+/// myEnumValue value;
+/// if (!kodi::CheckSettingEnum<myEnumValue>("my_enum_value", value))
+///   value = valueA; // My default of them
+/// ~~~~~~~~~~~~~
+///
+template<typename enumType>
+inline bool CheckSettingEnum(const std::string& settingName, enumType& settingValue)
+{
+  using namespace kodi::addon;
```

Comment:
Really? Never ever use 'using namespace' in a header file, as this can lead to all kind of weird errors when this header gets included together with other headers. Applies to newly added and existing code in this (and other addon header?) files. This needs to be fixed.

Reply:
Ooops, I somehow missed that the using scope is limited to the function. All good. Sorry for the noise.